core/build/build.go

```diff
@@ -0,0 +1,24 @@
+package build
+
+// The build module utilizes build tags to determine the environment that this binary was built to target
+// the currently supported build modes are dev, test. Setting both tags is not allowed and will result to compilation errors.
+
+type Build string
+
+const (
+	Prod Build = "prod"
+	Dev  Build = "dev"
+	Test Build = "test"
+)
+
+func DevelopmentBuild() bool {
+	return Mode == Dev
+}
+
+func TestBuild() bool {
+	return Mode == Test
+}
+
+func ProdBuild() bool {
+	return Mode == Prod
+}
```

Comment:
nit/ `build.XBuild()` is redundant. What about `Mode` instead?
```suggestion
func DevMode() bool {
	return Mode == Dev
}

func TestMode() bool {
	return Mode == Test
}

func ProdMode() bool {
	return Mode == Prod
}
```
Or maybe an `Is` prefix would be more idiomatic?
```suggestion
func IsDevMode() bool {
	return Mode == Dev
}

func IsTestMode() bool {
	return Mode == Test
}

func IsProdMode() bool {
	return Mode == Prod
}
```
I like s/Development/Dev too, but that seems less important :shrug: